Source/Modules/javascript.cxx

```diff
@@ -2725,24 +2738,42 @@ int TypeScriptTypes::classHandler(Node *n)
   int returnValue = Language::classHandler(n);
   if (generateTsTypes)
   {
-    if (!emittedImport) {
-      String *swigType = NewString("SWIGTYPE");
-      String *typescriptImports = typemapLookup(n, "typescriptimports", swigType);
-      Printf(declarationFilePtr, "%s\n", typescriptImports);
-      Delete(swigType);
-      Delete(typescriptImports);
-      emittedImport = true;
-    }
-
     tsTypeInterfaceDeclaration->setBaseClassName(getBaseClass(n));
-    tsTypeInterfaceDeclaration->typesFilePtr = declarationFilePtr;
-    tsTypeInterfaceDeclaration->generateTsTypes();
+    String *tmpInterfaceCode = tsTypeInterfaceDeclaration->emitTsTypes();
+    Append(interfaceCode,tmpInterfaceCode);
+    Delete(tmpInterfaceCode);
     delete tsTypeInterfaceDeclaration;
     tsTypeInterfaceDeclaration = NULL;
   }
   return returnValue;
 }
 
+/**
+ * Handler executed when a pragma is found in the Parse Tree
+ *
+ * @param n The node found that have the type %pragma
+ */
+int TypeScriptTypes::pragmaDirective(Node *n) {
+  if (generateTsTypes)
+  {
+    String *tsImportsLanguage = NewString("javascript");
+    String *tsImportsName = NewString("tsimports");
```

Comment:
Very minor, but I don't think you need `String`s for this, AFAIR `Strcmp()` works with `char*` strings directly too.